src/local_scheduler/local_scheduler_client.cc

```diff
@@ -10,30 +10,58 @@
 LocalSchedulerConnection *LocalSchedulerConnection_init(
     const char *local_scheduler_socket,
     ActorID actor_id,
-    bool is_worker) {
+    bool is_worker,
+    int64_t num_gpus) {
   LocalSchedulerConnection *result =
       (LocalSchedulerConnection *) malloc(sizeof(LocalSchedulerConnection));
   result->conn = connect_ipc_sock_retry(local_scheduler_socket, -1, -1);
+  result->actor_id = actor_id;
+  result->gpu_ids = new std::vector<int>();
```

Comment:
Let's make result->gpu_ids a vector and not a pointer to a vector. Then we don't need to call new and also don't need to delete it later. In this case, the malloc
```
LocalSchedulerConnection *result =
        (LocalSchedulerConnection *) malloc(sizeof(LocalSchedulerConnection));
```
needs to be done with new and freed with delete.